pkg/scheduler/core/generic_scheduler.go

```diff
@@ -630,19 +637,25 @@ func (g *genericScheduler) podFitsOnNode(
 	// they may end up getting scheduled to a different node.
 	for i := 0; i < 2; i++ {
 		metaToUse := meta
+		stateToUse := state
 		nodeInfoToUse := info
 		if i == 0 {
-			podsAdded, metaToUse, nodeInfoToUse = addNominatedPods(pod, meta, info, queue)
+			err := error(nil)
+			podsAdded, metaToUse, stateToUse, nodeInfoToUse, err = g.addNominatedPods(pod, meta, state, info, queue)
+			if err != nil {
+				return false, []predicates.PredicateFailureReason{}, nil, err
+			}
 		} else if !podsAdded || len(failedPredicates) != 0 {
 			break
 		}
+
 		for _, predicateKey := range predicates.Ordering() {
 			var (
 				fit     bool
 				reasons []predicates.PredicateFailureReason
 				err     error
 			)
-			//TODO (yastij) : compute average predicate restrictiveness to export it as Prometheus metric
```

Comment:
Why is the above line removed?

Reply:
It doesn't really make sense to compute this in the scheduler directly, we will be exporting a metric that tracks filtering reasons, and from that one can compute any summary statistic they want.